### PR TITLE
[@types/node] add missing optional code argument to util.deprecate

### PR DESCRIPTION
**Pull request by sokra**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/util.html#util_util_deprecate_fn_msg_code
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by r3nya — APPROVED** [review]

**Review by Flarna — APPROVED** [review]

**Review by jamesgeorge007 — APPROVED** [review]

**Review by mohsen1 — APPROVED** [review]

File: types/node/util.d.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ declare module "util" {
     function isSymbol(object: any): object is symbol;
     /** @deprecated since v4.0.0 - use `value === undefined` instead. */
     function isUndefined(object: any): object is undefined;
-    function deprecate<T extends Function>(fn: T, message: string): T;
+    function deprecate<T extends Function>(fn: T, message: string, code?: string): T;
```
Comment (trivikr) [inline types/node/util.d.ts]: This was added in v0.8.0 https://nodejs.org/api/util.html#util_util_deprecate_fn_msg_code

Can you please make changes in other existing supported versions too?
* 10.x https://github.com/DefinitelyTyped/DefinitelyTyped/blob/aee73c479bd3f156b1468f3fd7e34811263f9e1a/types/node/v10/util.d.ts#L58
* 8.x https://github.com/DefinitelyTyped/DefinitelyTyped/blob/aee73c479bd3f156b1468f3fd7e34811263f9e1a/types/node/v8/base.d.ts#L5661

**Review by trivikr — CHANGES_REQUESTED** [review]
Selecting "Request changes" to trigger TypeScript Bot

**Review by a-tarasyuk — APPROVED** [review]

File: types/node/util.d.ts (hunk 1 above)
Comment (sokra) [reply types/node/util.d.ts]: Done

File: types/node/util.d.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ declare module "util" {
     function isSymbol(object: any): object is symbol;
     /** @deprecated since v4.0.0 - use `value === undefined` instead. */
     function isUndefined(object: any): object is undefined;
-    function deprecate<T extends Function>(fn: T, message: string): T;
+    function deprecate<T extends Function>(fn: T, message: string, code?: string | undefined): T;
     function isDeepStrictEqual(val1: any, val2: any): boolean;
```
Comment (SimonSchick) [inline types/node/util.d.ts]: Using `?` and `| undefined` are redundant, just keep the `?`.

File: types/node/test/util.ts [hunk 3]
```diff
@@ -127,8 +127,10 @@ import { readFile } from 'fs';
     const foo = () => {};
     // $ExpectType () => void
     util.deprecate(foo, 'foo() is deprecated, use bar() instead');
-    // $ExpectType <T extends Function>(fn: T, message: string) => T
+    // $ExpectType <T extends Function>(fn: T, message: string, code?: string | undefined) => T
```
Comment (nguymin4) [inline types/node/test/util.ts]: It should be changed here and below as well.

**Review by nguymin4 — CHANGES_REQUESTED** [review]

**Comment by sokra** [comment]
I think this is ready to be reviewed again

File: types/node/test/util.ts (hunk 3 above)
Comment (trivikr) [reply types/node/test/util.ts]: @sokra update this comment?

**Review by SimonSchick — APPROVED** [review]

File: types/node/test/util.ts (hunk 3 above)
Comment (sokra) [reply types/node/test/util.ts]: Changing it to `// $ExpectType <T extends Function>(fn: T, message: string, code?: string) => T` won't work:

``` text
Error in node
Error: C:/Repos/DefinitelyTyped/types/node/test/util.ts:131:5
ERROR: 131:5  expect  TypeScript@3.1 expected type to be:
  <T extends Function>(fn: T, message: string, code?: string) => T
got:
  <T extends Function>(fn: T, message: string, code?: string | undefined) => T
ERROR: 133:5  expect  TypeScript@3.1 expected type to be:
  <T extends Function>(fn: T, message: string, code?: string) => T
got:
  <T extends Function>(fn: T, message: string, code?: string | undefined) => T
```

I guess typescript internally adds `| undefined` for optional arguments, so the expected type need to include this too.

The type is defined without `| undefined`:

``` ts
function deprecate<T extends Function>(fn: T, message: string, code?: string): T;
```

Comment (nguymin4) [reply types/node/test/util.ts]: I just want to confirm again so there are differences between tests for `latest` and `v10`
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38753/files#diff-2d1106145a2d470a4a929f463c4a0292R949

If that's the case then I can approve it.

Comment (nguymin4) [reply types/node/test/util.ts]: If other contributors don't have any other reviews then it's ok for me.

**Review by nguymin4 — APPROVED** [review]

**Review by trivikr — APPROVED** [review]

**Pull request merged**